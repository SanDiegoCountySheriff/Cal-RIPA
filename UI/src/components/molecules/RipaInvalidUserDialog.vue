<template>
  <v-dialog v-model="model" max-width="400px" :light="getLight" :dark="getDark">
    <v-card>
      <v-card-title>
        <span>User Not Authorized</span>
      </v-card-title>

      <v-card-text>
        <p>You are not authorized to use this application.</p>
        <p>
          Please check with your administrator to ensure you have the correct
          permissions.
        </p>
      </v-card-text>
    </v-card>
  </v-dialog>
</template>

<script>
export default {
  name: 'ripa-invalid-user-dialog',

  data() {
    return {
      viewModel: this.showDialog,
    }
  },

  computed: {
    model: {
      get() {
        return this.viewModel
      },
      set(newValue) {
        if (!newValue) {
          if (this.onClose) {
            this.onClose()
          }
        }
        this.viewModel = newValue
      },
    },

    getLight() {
      return this.$vuetify.theme.dark
    },

    getDark() {
      return !this.$vuetify.theme.dark
    },
  },

  methods: {
    handleClose() {
      if (this.onClose) {
        this.onClose()
      }
    },
  },

  watch: {
    showDialog(newValue) {
      this.viewModel = newValue
    },
  },

  props: {
    showDialog: {
      type: Boolean,
      default: false,
    },
    onClose: {
      type: Function,
      default: () => {},
    },
  },
}
</script>

<style lang="scss">
.v-dialog:not(.v-dialog--fullscreen) {
  max-height: 600px !important;
}
</style>
