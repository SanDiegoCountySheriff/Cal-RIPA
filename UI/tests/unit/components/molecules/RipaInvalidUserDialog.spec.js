import RipaInvalidUserDialog from '@/components/molecules/RipaInvalidUserDialog.vue'
import { shallowMount, mount } from '@vue/test-utils'
import Vuetify from 'vuetify'

describe('Ripa Invalid User Dialog', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const factory = propsData => {
    return shallowMount(RipaInvalidUserDialog, {
      vuetify,
      propsData: {
        ...propsData,
      },
    })
  }

  it('should match snapshot', () => {
    wrapper = mount(RipaInvalidUserDialog, { vuetify })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
