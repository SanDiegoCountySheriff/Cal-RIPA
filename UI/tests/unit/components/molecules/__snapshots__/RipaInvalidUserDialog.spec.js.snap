// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Invalid User Dialog should match snapshot 1`] = `
<div role="dialog" class="v-dialog__container">
  <!---->
</div>
`;
